File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/masterdata/MasterDataRepositoryIntegrationTest.java
package io.zenwave360.example.clinicaltool.modules.masterdata;

import io.zenwave360.example.clinicaltool.common.BaseRepositoryIntegrationTest;
import io.zenwave360.example.clinicaltool.modules.masterdata.domain.*;
import jakarta.persistence.EntityManager;
import java.time.*;
import java.util.List;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

class MasterDataRepositoryIntegrationTest extends BaseRepositoryIntegrationTest {

    @Autowired
    EntityManager entityManager;

    @Autowired
    MasterDataRepository masterDataRepository;

    @Test
    void findAllTest() {
        var results = masterDataRepository.findAll();
        Assertions.assertFalse(results.isEmpty());
    }

    @Test
    void findByIdTest() {
        var id = 1L;
        var masterData = masterDataRepository.findById(id).orElseThrow();
        Assertions.assertNotNull(masterData.getId());
        Assertions.assertNotNull(masterData.getVersion());
    }

    @Test
    void saveTest() {
        MasterData masterData = new MasterData();
        masterData.setType(MasterDataType.values()[0]);
        masterData.setKey("");
        masterData.setValue("");
        masterData.setTranslations(List.of(new MasterDataTranslation()));

        // Persist aggregate root
        var created = masterDataRepository.save(masterData);

        // reloading to get relationships persisted by id
        entityManager.flush();
        entityManager.refresh(created);
        Assertions.assertNotNull(created.getId());
        Assertions.assertNotNull(created.getVersion());
    }

    @Test
    void updateTest() {
        var id = 1L;
        var masterData = masterDataRepository.findById(id).orElseThrow();
        masterData.setType(MasterDataType.values()[0]);
        masterData.setKey("");
        masterData.setValue("");
        masterData.setTranslations(List.of(new MasterDataTranslation()));

        masterData = masterDataRepository.save(masterData);
        Assertions.assertEquals(MasterDataType.values()[0], masterData.getType());
        Assertions.assertEquals("", masterData.getKey());
        Assertions.assertEquals("", masterData.getValue());
        //        Assertions.assertEquals(List.of(new MasterDataTranslation()), masterData.getTranslations());
    }

    @Test
    void deleteTest() {
        var id = 1L;
        masterDataRepository.deleteById(id);
        var notFound = masterDataRepository.findById(id);
        Assertions.assertFalse(notFound.isPresent());
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/masterdata/MasterDataRepositoryIntegrationTest.kt
package io.zenwave360.example.clinicaltool.modules.masterdata

import io.zenwave360.example.clinicaltool.common.BaseRepositoryIntegrationTest
import io.zenwave360.example.clinicaltool.modules.masterdata.domain.*
import jakarta.persistence.EntityManager
import java.time.*
import java.util.List
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.repository.findByIdOrNull

class MasterDataRepositoryIntegrationTest : BaseRepositoryIntegrationTest() {

    @Autowired lateinit var entityManager: EntityManager

    @Autowired lateinit var masterDataRepository: MasterDataRepository

    @Test
    fun findAllTest() {
        val results = masterDataRepository.findAll()
        Assertions.assertFalse(results.isEmpty())
    }

    @Test
    fun findByIdTest() {
        val id = 1L
        val masterData =
            masterDataRepository.findByIdOrNull(id) ?: throw NoSuchElementException(" not found with id: $id")
        Assertions.assertNotNull(masterData.id)
        Assertions.assertNotNull(masterData.version)
    }

    @Test
    fun saveTest() {
        val masterData = MasterData()
        masterData.type = MasterDataType.values()[0]
        masterData.key = ""
        masterData.value = ""
        masterData.translations = mutableListOf(MasterDataTranslation())

        // Persist aggregate root
        val created = masterDataRepository.save(masterData)

        // reloading to get relationships persisted by id
        entityManager.flush()
        entityManager.refresh(created)
        Assertions.assertNotNull(created.id)
        Assertions.assertNotNull(created.version)
    }

    @Test
    fun updateTest() {
        val id = 1L
        val masterData =
            masterDataRepository.findByIdOrNull(id) ?: throw NoSuchElementException(" not found with id: $id")
        masterData.type = MasterDataType.values()[0]
        masterData.key = ""
        masterData.value = ""
        masterData.translations = mutableListOf(MasterDataTranslation())

        val updated = masterDataRepository.save(masterData)
        Assertions.assertEquals(MasterDataType.values()[0], updated.type)
        Assertions.assertEquals("", updated.key)
        Assertions.assertEquals("", updated.value)
//        Assertions.assertEquals(mutableListOf(MasterDataTranslation()), updated.translations)
    }

    @Test
    fun deleteTest() {
        val id = 1L
        masterDataRepository.deleteById(id)
        val notFound = masterDataRepository.findByIdOrNull(id)
        Assertions.assertNull(notFound)
    }
}
